### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by Doheera-kosi**
# Milestone reached: 
## Objective:
- Wrote unit tests with React Testing Library.
- Wrote unit tests with snapshots with Jest.

## General requirements
- Made sure that there are no linter errors
- Made sure that we used correct Gitflow
- Made sure that we documented our work in a professional way.

## JavaScript requirements
Followed our list of best practices for JavaScript

## Project requirements
We added tests to the Math Magicians app using [Jest](https://jestjs.io/) and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro). Take a look at their documentation.

- Set up React Testing Library.
- Followed the instructions in the official documentation
- Wrote unit tests for the files operate.js and calculate.js using Jest.
- Created unit tests for all React components:
- Used Jest snapshots to test the components.
- Used React Testing Library to simulate user interaction.
- Run full tests suite using Jest. All tests should pass.

### CODE REVIEW

File: src/unit_test/CalculatorSnapshots.test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import renderer from 'react-test-renderer';
+import React from 'react';
+import Calculator from '../components/Calculator';
+
+it('Calculator renders correctly', () => {
+  const tree = renderer
+    .create(<Calculator />)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (DammyShittu) [inline src/unit_test/CalculatorSnapshots.test.js:10]: - [ ] Using the react testing library you are required to write tests that simulate user interactions like clicking buttons and performing calculations. Kindly add such tests.

  You can visit this basic tutorial for clues on how to do that: [https://www.robinwieruch.de/react-testing-library/](https://www.robinwieruch.de/react-testing-library/)

- [x] One of your tests is failing. Kindly fix that. I think this is coming from the `App.test.js` file.

  ieoe

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Doheera-kosi @jaamanu,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/unit_test/CalculatorSnapshots.test.js (hunk 1 above)
Comment (Doheera-kosi) [reply src/unit_test/CalculatorSnapshots.test.js:10]: Thank you! Issues resolved

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Doheera-kosi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**